controller/users.go

```diff
@@ -77,6 +78,130 @@ func (c *UsersController) Show(ctx *app.ShowUsersContext) error {
 	})
 }
 
+func isServiceAccount(ctx context.Context) (bool, error) {
+	tokenManager, err := token.ReadManagerFromContext(ctx)
+	if err != nil {
+		return false, err
+	}
+	return (*tokenManager).IsServiceAccount(ctx), nil
+}
+
+// CreateUserAsServiceAccount creates a user when requested using a service account token
+func (c *UsersController) CreateUserAsServiceAccount(ctx *app.CreateUserAsServiceAccountUsersContext) error {
+
+	isSvcAccount, err := isServiceAccount(ctx)
+	if err != nil {
+		log.Error(ctx, map[string]interface{}{
+			"err": err,
+		}, "failed to determine if account is a service account")
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrUnauthorized(err))
+
+	}
+	if !isSvcAccount {
+		log.Error(ctx, map[string]interface{}{
+			"identity_id": ctx.ID,
+		}, "account used to call create api is not a service account")
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrUnauthorized(errs.New("a non-service account tried to create a user.")))
```

Comment:
If there is no token at all then Goa will generate 401. Which is correct.
But if the token doesn't have permissions to do this operation then it would be better to return 403 (Forbidden). And we don't have to be too specific in the error message we return to the client. Add details to the error log message instead.
Generally speaking 401 means we don't know who you are. And 403 means we know you but you don't have permissions. If I understand it correctly ;)

Reply:
Thanks!
7b462ed

I am usually undecided between
1) Forbidden or Unauthorized 
2) always say Unauthorized, that way we don't divulge information to the attacker. 

In case of (1)  "you are close, you got the right kind of token, but your identity is not as we expect it to be" :) 

but I understand if our services need to communicate with each other properly we need to be clear with the responses.